contracts/src/main/kotlin/com/contractsAndStates/states/RoadState.kt

```diff
@@ -45,3 +45,153 @@ data class RoadState(
     }
 }
 
+/**
+ * Returns the player who should received the Longest Road Card given the current state
+ * of the board, roads, settlements, players and the current holder of the card. Either the
+ * current holder and the return value can be null meaning no player had or will have the card.
+ */
+fun longestRoad(board: PlacedHexTiles,
+                roads: List<RoadState>,
+                settlements: List<SettlementState>,
+                players: List<Party>,
+                currentHolder: Party?): Party? {
+    val candidates = longestRoadsForAllPlayers(board, roads, settlements, players)
+    return assignLongestRoad(currentHolder, candidates)
+}
+
+data class LongestRoadCandidate(val player: Party, val longestRoadLength: Int)
+
+/**
+ * Given a list of pairs of parties and count of the longest road and the current player with
+ * the longest road card (null means no one has the card), return the new card holder. It can
+ * be a new player, the same player or null if no one meets the criteria.
+ */
+fun assignLongestRoad(holder: Party?, candidates: List<LongestRoadCandidate>): Party? {
+
+    val orderedCandidates = candidates.sortedByDescending { it.longestRoadLength }
+
+    // Longest candidate does not have at least 5 roads
+    if (orderedCandidates[0].longestRoadLength < 5)
```

Comment:
This `5` should be made as a `const` in the `companion object`.

Reply:
Ok

Reply:
Agreed - we've been using this as best practice. (see gameboardstate)